### DISCUSSION TITLE
Date picker is not closing when it is opened while horizontal scrolling, bug?

### DISCUSSION THREAD

**Discussion by ch3pe** [ghiscoding/Angular-Slickgrid#765] [Q&A]
![image](https://user-images.githubusercontent.com/54966395/119097136-04682f80-ba32-11eb-82cc-5b22e0371936.png)
So ,
When i click on the datepicker in which there is horizontal scroll enabled, i use my mouse trackpad to scroll horizontally , the datepicker moves with the grid to left and right. It should close when we scroll left/right.

This issue only happens with the trackpad / horizontal wheel scrolling  , if you click the scroll bar and drag it , then the date picker closes.
How do i make it close ?

**Comment by ghiscoding** [comment]
not a bug but if you really want to close it, then I guess you could try with the [onScroll](https://github.com/6pac/SlickGrid/wiki/Grid-Events) event

**Reply by ch3pe** [reply]
I wrote something like : 
`onScroll(){ 
var el = document.querySelector(".slick-viewport-right")
if(el.scrollLeft) {
// remove class
}
}` with no success.

**Reply by ghiscoding** [reply]
That's not the way to use the SlickGrid events, see this [Wiki](https://github.com/ghiscoding/Angular-Slickgrid/wiki/Grid-&-DataView-Events#example-with-event-dispatch-sgonx)  `sgOnScroll`

**Reply by ch3pe** [reply]
I dont understand how to use sgOnScroll? The parameters to onScroll says scrollLeft, scrollTop , Can i use these properties directly in my function?
I have added this - 
`(sgOnScroll)="onScroll($event.detail.eventData, $event.detail.args)"`

and decorated onScroll method as this - 
` @HostListener(['$event', '$event.detail.eventData', '$event.detail.args'])
  onScroll(event) { //how to use this?}` 

Now whatever i want to "do" when a scroll event occurs inside the slickgrid ( horizontal / vertical ) , I can write in this method? Can you tell me how can I use scrollLeft and scrollTop ? 

EDIT: I just added `$(".flatpickr-calendar").removeClass("open");` in the method and the calendar is disappearing onscroll,
But the problem is , it is not clickable again once the user scrolls and tries to click on it again. ( It is only clickable when user clicks anywhere else then clicks on the calendar, 
So this is happening because i'm just removing the "open" class and hiding the UI , but the calendar (flatpickr) is still active ( i inspected and yes it was staying active) , 
Any idea how to close the flatpickr as a whole when the user scrolls?
Thanks for replying!

**Reply by ghiscoding** [reply]
@ch3pe  I know you wrote this a month ago but I just saw your edit, the code you use is probably not ideal. I think you can simply query the date picker DOM element and call the `close()` method from the Flatpickr instance
```ts
const fp = document.querySelector("#myInput")._flatpickr;
fp.close();
```
Just take a look at [Flapickr docs](https://flatpickr.js.org/instance-methods-properties-elements/)
